### ISSUE TITLE
E0277 should put emphasis on containing function's return type

### ISSUE THREAD

**Issue by mibac138** [rust-lang/rust#63078]
I got the following error today:
```
error[E0277]: the `?` operator can only be used in a function that returns `Result` or `Option` (or another type that implements `std::ops::Try`)
  --> src/main.rs:29:20
   |
29 |     let mut resp = reqwest::get("...")?;
   |                    ^^^^^^^^^^^^^^^^^^^^ cannot use the `?` operator in a function that returns `()`
   |
   = help: the trait `std::ops::Try` is not implemented for `()`
   = note: required by `std::ops::Try::from_error`
```
I was confused for a minute as I misinterpreted what the error was saying. I double-checked that `reqwest::get` does indeed return a `Result` but a moment later I realized that it's the function that's containing `reqwest::get` I should be looking at which indeed did return `()` after which I fixed the issue. 

I believe this error message could be somehow improved and/or worded differently to put more stress on the containing function - after all, that's what I should've looked at (maybe the containing function should be even shown in the error message like `reqwest::get` is?) though I can see how it can not always be the case. However, I do still think both the containing method's return type _and_ the `reqwest::get` call should be highlighted for the best experience and to easily allow a full understanding of what's happening.

**Comment by fakenine** [comment]
Hello! I'd like to work on this one.

**Comment by ayazhafiz** [comment]
@fakenine are you still interested in this?

**Comment by fakenine** [comment]
@hafiz Sorry, I forgot about this issue! You can take it if you want. I don’t think I will have the time to address it at the moment.

**Comment by chandrikamutalik** [comment]
Hello! I'd like to work on this if it's not being worked on already. @fakenine , @ayazhafiz can you guys please tell me if either of you are working on this! :)

**Comment by ayazhafiz** [comment]
I'm not :)

**Comment by reese** [comment]
As mentioned by @estebank, I think this issue is closed by #66651, which updated this error message with some more emphasis on the return type (see [here](https://github.com/rust-lang/rust/pull/66651/files#diff-3dcd3c64b849a3f377c577aa187b3309R16)).

**Comment by estebank** [comment]
Closing, since #66651 landed in nightly [the output is now](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=fbf4a28f7957195e199a755fa116af11)

```
error[E0277]: the `?` operator can only be used in a function that returns `Result` or `Option` (or another type that implements `std::ops::Try`)
 --> src/lib.rs:2:5
  |
1 | / fn bar() {
2 | |     foo()?;
  | |     ^^^^^^ cannot use the `?` operator in a function that returns `()`
3 | | }
  | |_- this function should return `Result` or `Option` to accept `?`
  |
  = help: the trait `std::ops::Try` is not implemented for `()`
  = note: required by `std::ops::Try::from_error`
```
